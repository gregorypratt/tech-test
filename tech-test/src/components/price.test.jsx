import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Price from './price';

describe('Price', () => {
  const props = { decimal: '9', num: '8', den: '1', viewAsDecimal: true };

  it('renders without crashing', () => {
    const component = shallow(<Price.WrappedComponent />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('shows the decimal value', () => {
    const component = shallow(<Price.WrappedComponent {...props} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('shows the fractional value', () => {
    props.viewAsDecimal = false;
    const component = shallow(<Price.WrappedComponent {...props} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
