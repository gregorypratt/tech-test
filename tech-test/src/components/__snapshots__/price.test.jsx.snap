// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price renders without crashing 1`] = `
<span>
  -
</span>
`;

exports[`Price shows the decimal value 1`] = `
<span>
  9.00
</span>
`;

exports[`Price shows the fractional value 1`] = `
<span>
  8/1
</span>
`;
